src/gebt_poc/static_beam_element.cpp

```diff
@@ -0,0 +1,253 @@
+    );
+    CalculateStaticIterationMatrix(
+        position_vectors_, gen_coords_1D, stiffness_matrix_, quadrature_, quadrant_1
+    );
+    KokkosBlas::gemm("N", "N", 1.0, quadrant_1, tangent_operator, 0.0, quadrant_1);
+
+    // quadrant_2 = Transpose([B(q)])
+    auto quadrant_2 = Kokkos::subview(
+        iteration_matrix, Kokkos::make_pair(zero, size_dofs),
+        Kokkos::make_pair(size_dofs, size_iteration)
+    );
+    auto constraints_gradient_matrix =
+        Kokkos::View<double**>("constraints_gradient_matrix", size_constraints, size_dofs);
+    ConstraintsGradientMatrix(gen_coords_1D, position_vectors_, constraints_gradient_matrix);
+    // TODO ** Question for reviewers **
+    // How to transpose a matrix using KokkosBlas?
+    auto temp = gen_alpha_solver::transpose_matrix(constraints_gradient_matrix);
+    Kokkos::deep_copy(quadrant_2, temp);
+
+    // quadrant_3 = B(q) * T(h dq)
+    auto quadrant_3 = Kokkos::subview(
+        iteration_matrix, Kokkos::make_pair(size_dofs, size_iteration),
+        Kokkos::make_pair(zero, size_dofs)
+    );
+    KokkosBlas::gemm("N", "N", 1.0, constraints_gradient_matrix, tangent_operator, 0.0, quadrant_3);
+
+    return iteration_matrix;
+}
+
+void StaticBeamLinearizationParameters::TangentOperator(
+    const Kokkos::View<double*> psi, Kokkos::View<double**> tangent_operator
+) {
+    // TODO ** Question for reviewers **
+    // Any better way to create the identity matrix than parallel_for?
+    auto populate_matrix = KOKKOS_LAMBDA(size_t) {
+        tangent_operator(0, 0) = 1.;
+        tangent_operator(1, 1) = 1.;
+        tangent_operator(2, 2) = 1.;
+        tangent_operator(3, 3) = 1.;
+        tangent_operator(4, 4) = 1.;
+        tangent_operator(5, 5) = 1.;
+    };
+    Kokkos::parallel_for(1, populate_matrix);
+
+    auto psi_0 = Kokkos::subview(psi, Kokkos::make_pair(0, 1));
+    auto psi_1 = Kokkos::subview(psi, Kokkos::make_pair(1, 2));
+    auto psi_2 = Kokkos::subview(psi, Kokkos::make_pair(2, 3));
+    const double phi = std::sqrt(psi_0(0) * psi_0(0) + psi_1(0) * psi_1(0) + psi_2(0) * psi_2(0));
+
+    if (std::abs(phi) > kTolerance) {
+        auto psi_cross_prod_matrix = gen_alpha_solver::create_cross_product_matrix(psi);
+        auto psi_times_psi = Kokkos::View<double**>("psi_times_psi", 3, 3);
+        KokkosBlas::gemm(
+            "N", "N", 1.0, psi_cross_prod_matrix, psi_cross_prod_matrix, 0.0, psi_times_psi
+        );
+
+        auto quadrant4 =
+            Kokkos::subview(tangent_operator, Kokkos::make_pair(3, 6), Kokkos::make_pair(3, 6));
+        auto temp = Kokkos::View<double**>("temp", 3, 3);
+        KokkosBlas::scal(temp, (std::cos(phi) - 1.0) / (phi * phi), psi_cross_prod_matrix);
+        KokkosBlas::axpy(1.0, temp, quadrant4);
+        KokkosBlas::scal(temp, (1.0 - std::sin(phi) / phi) / (phi * phi), psi_times_psi);
+        KokkosBlas::axpy(1.0, temp, quadrant4);
```

Comment:
If we wrote this as a single parallel_for, would we still need to allocate the temp view?